src/ol_infrastructure/applications/superset/__main__.py

```diff
@@ -0,0 +1,742 @@
+superset_web_tag_specs = [
+    TagSpecification(
+        resource_type="instance",
+        tags=aws_config.merged_tags({"Name": f"superset-web-{stack_info.env_suffix}"}),
+    ),
+    TagSpecification(
+        resource_type="volume",
+        tags=aws_config.merged_tags({"Name": f"superset-web-{stack_info.env_suffix}"}),
+    ),
+]
+
+superset_web_lt_config = OLLaunchTemplateConfig(
+    block_device_mappings=superset_web_block_device_mappings,
+    image_id=superset_ami.id,
+    instance_type=superset_config.get("instance_type")
+    or InstanceTypes.burstable_medium,
+    instance_profile_arn=superset_profile.arn,
+    security_groups=[
+        superset_security_group.id,
+        consul_security_groups["consul_agent"],
+        data_vpc["security_groups"]["web"],
+    ],
+    tags=aws_config.merged_tags({"Name": f"superset-web-{stack_info.env_suffix}"}),
+    tag_specifications=superset_web_tag_specs,
+    user_data=consul_datacenter.apply(
+        lambda consul_dc: base64.b64encode(
+            "#cloud-config\n{}".format(
+                yaml.dump(
+                    {
+                        "write_files": [
+                            {
+                                "path": "/etc/consul.d/02-autojoin.json",
+                                "content": json.dumps(
+                                    {
+                                        "retry_join": [
+                                            "provider=aws tag_key=consul_env "
+                                            f"tag_value={consul_dc}"
+                                        ],
+                                        "datacenter": consul_dc,
+                                    }
+                                ),
+                                "owner": "consul:consul",
+                            },
+                            {
+                                "path": "/etc/default/vector",
+                                "content": textwrap.dedent(
+                                    f"""\
+                            ENVIRONMENT={consul_dc}
+                            APPLICATION=superset
+                            SERVICE=data-platform
+                            VECTOR_CONFIG_DIR=/etc/vector/
+                            AWS_REGION={aws_config.region}
+                            GRAFANA_CLOUD_API_KEY={grafana_credentials['api_key']}
+                            GRAFANA_CLOUD_PROMETHEUS_API_USER={grafana_credentials['prometheus_user_id']}
+                            GRAFANA_CLOUD_LOKI_API_USER={grafana_credentials['loki_user_id']}
+                            """
+                                ),
+                                "owner": "root:root",
+                            },
+                            {
+                                "path": "/etc/default/superset",
+                                "content": f"DOMAIN={superset_domain}",
+                            },
```

Comment:
I don't see where this is ever used?

Reply:
It's not. Instead it's adding the domain to the .env file directly. I'll remove this to avoid confusion.